seed_data/lib.py

```diff
@@ -170,102 +178,97 @@ def set_edx_key(cls, obj, course_run):
-        obj.data.update({
-            'user': self.user.username,
-            'mode': 'verified' if verified else 'not verified'
-        })
-        obj.data['course_details'] = obj.data.get('course_details', {})
-        obj.data['course_details'].update({
-            'course_id': course_run.edx_course_key,
-            'enrollment_start': course_run.enrollment_start.isoformat(),
-            'enrollment_end': course_run.enrollment_end.isoformat()
-        })
-
-    def set_or_create(self, *args, **kwargs):
-        """Updates the specific cache timestamp before taking care of the cached object"""
-        defaults = {'user': self.user, 'enrollment': datetime.utcnow()}
-        UserCacheRefreshTime.objects.update_or_create(user=self.user, defaults=defaults)
-        return super().set_or_create(*args, **kwargs)
+    @model_obj_editor
+    def build_data_property(self, course_run, verified=True, **kwargs):
+        return {
+            'user': self.username,
+            'mode': 'verified' if verified else 'not verified',
+            'is_active': True,
+            'course_details': {
+                'course_id': course_run.edx_course_key,
+                'enrollment_start': course_run.enrollment_start.isoformat(),
+                'enrollment_end': course_run.enrollment_end.isoformat()
+            }
+        }
 
 
 class CachedCertificateHandler(CachedHandler):
     """Provides functionality to CachedCertificate objects"""
     model_cls = CachedCertificate
+    cache_type = 'certificate'
 
     @classmethod
     def set_edx_key(cls, obj, course_run):
         obj.data['course_id'] = course_run.edx_course_key
 
-    @cached_obj_editor
-    def set_data(self, obj, course_run, grade=DEFAULT_GRADE, **kwargs):
-        obj.data.update({
-            'username': self.user.username,
+    @model_obj_editor
+    def build_data_property(self, course_run, grade=DEFAULT_GRADE, **kwargs):
+        return {
+            'username': self.username,
             'course_id': course_run.edx_course_key,
             'certificate_type': 'verified',
             'grade': str(grade)
-        })
-
-    def set_or_create(self, *args, **kwargs):
-        """Updates the specific cache timestamp before taking care of the cached object"""
-        defaults = {'user': self.user, 'certificate': datetime.utcnow()}
-        UserCacheRefreshTime.objects.update_or_create(user=self.user, defaults=defaults)
-        return super().set_or_create(*args, **kwargs)
+        }
 
 
 class CachedCurrentGradeHandler(CachedHandler):
     """Provides functionality to CachedCurrentGrade objects"""
     model_cls = CachedCurrentGrade
+    cache_type = 'current_grade'
 
     @classmethod
     def set_edx_key(cls, obj, course_run):
         obj.data['course_key'] = course_run.edx_course_key
 
-    @cached_obj_editor
-    def set_data(self, obj, course_run, grade=DEFAULT_GRADE, **kwargs):
-        obj.data.update({
+    @model_obj_editor
```

Comment:
I'm a little bit confused on the usage of `model_obj_editor`. In the usages I see in this PR, the wrapped methods don't modify the model. In that case, if the model really does need to be saved, should that be done elsewhere instead of here?